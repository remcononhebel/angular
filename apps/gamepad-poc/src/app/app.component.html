<h1>Hello, World!</h1>

<div>
  <app-gamepad-poc></app-gamepad-poc>
</div>
