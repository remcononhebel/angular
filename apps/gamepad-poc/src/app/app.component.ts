import { Component } from '@angular/core';
import { GamepadPocComponent } from './gamepad-poc/gamepad-poc.component';

export interface JoystickViewModel {}

@Component({
  standalone: true,
  imports: [GamepadPocComponent],
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrl: './app.component.scss',
})
export class AppComponent {}
