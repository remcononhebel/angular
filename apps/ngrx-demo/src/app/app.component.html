<div class="app">
  <div class="app__available-books">
    <h1>Available books ({{ numberOfAvailableBooks$ | async }})</h1>
    <ngrx-demo-book-list
      class="book-list"
      [books]="(books$ | async)!"
      (add)="onAdd($event)"></ngrx-demo-book-list>
    <div class="app__action">
      <button
        (click)="selectAll()">
        Add all to collection
      </button>
    </div>
  </div>

  <div class="app__my-collection">
    <h1>My Collection ({{ numberOfBooksInCollections$ | async }})</h1>
    <ngrx-demo-book-collection
      class="book-collection"
      [books]="(bookCollection$ | async)!"
      (remove)="onRemove($event)">
    </ngrx-demo-book-collection>
    <div class="app__action">
      <button
        (click)="deselectAll()">
        Remove all from collection
      </button>
    </div>
  </div>
</div>
