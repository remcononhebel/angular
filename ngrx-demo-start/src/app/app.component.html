<h2>Available books</h2>
Hier komen de boeken
