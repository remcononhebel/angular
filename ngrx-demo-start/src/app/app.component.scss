.app {
  display: flex;
  flex-direction: row;

  &__available-books {
    width: 400px;
  }
}
