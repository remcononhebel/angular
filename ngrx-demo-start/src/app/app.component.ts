import { Component, OnInit } from '@angular/core';
import { Store } from '@ngrx/store';
import { Book } from './model/book.model';
import { BooksApiService } from './service/books-api.service';
import { BooksApiActions } from './state/books/books.actions';
import { selectAvailableBooks, selectNumberOfAvailableBooks } from './state/books/books.selectors';
import { CollectionActions } from './state/collection/collection.actions';
import { selectBookCollection, selectNumberOfBooksInCollection } from './state/collection/collection.selectors';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent implements OnInit {
  title = 'ngrx-demo-start';

  books$ = this.store.select(selectAvailableBooks);
  numberOfAvailableBooks$ = this.store.select(selectNumberOfAvailableBooks);

  bookCollection$ = this.store.select(selectBookCollection);
  numberOfBooksInCollections$ = this.store.select(selectNumberOfBooksInCollection);

  constructor(private readonly store: Store, private readonly booksApiService: BooksApiService) {}

  ngOnInit(): void {
    this.booksApiService.getBooks().subscribe((books: Book[]) => {
      this.store.dispatch(BooksApiActions.retrievedBookList({ books }));
    });
  }

  onAdd(bookId: string) {
    this.store.dispatch(CollectionActions.addBook({ bookId }));
  }

  onRemove(bookId: string) {
    this.store.dispatch(CollectionActions.removeBook({ bookId }));
  }

  selectAll() {
    this.store.select(selectAvailableBooks).subscribe((books) => {
      if (books.length > 0) {
        const bookIds = books.map(({ id }) => id);
        this.store.dispatch(CollectionActions.selectBooks({ bookIds }));
      }
    });
  }

  deselectAll() {
    this.store.dispatch(CollectionActions.deselectAllBooks());
  }
}
