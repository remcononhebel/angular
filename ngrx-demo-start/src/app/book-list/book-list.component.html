<div class="book-list">
  <div class="book-list__item" *ngFor="let book of books">
    <h2 class="book-list__item-title">
      {{ book.volumeInfo.title }}
    </h2>
    <div class="book-list__item-authors">by {{ book.volumeInfo.authors }}</div>
    <div class="book-list__item-add">
      <button (click)="add.emit(book.id)" data-test="add-button">
        Add to Collection
      </button>
    </div>
  </div>
</div>
