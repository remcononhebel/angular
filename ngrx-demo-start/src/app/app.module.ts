import { HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { StoreModule } from '@ngrx/store';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';

import { AppComponent } from './app.component';
import { BookCollectionComponent } from './book-collection/book-collection.component';
import { BookListComponent } from './book-list/book-list.component';
import { booksReducer } from './state/books/books.reducer';
import { collectionReducer } from './state/collection/collection.reducer';

@NgModule({
  declarations: [AppComponent, BookListComponent, BookCollectionComponent],
  imports: [
    BrowserModule,
    StoreModule.forRoot({
      books: booksReducer,
      collection: collectionReducer,
    }),
    HttpClientModule,
    StoreDevtoolsModule.instrument(),
  ],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
