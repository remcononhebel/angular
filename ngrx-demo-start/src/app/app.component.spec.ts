import { MockComponent } from 'ng-mocks';
import { Spectator, createComponentFactory } from '@ngneat/spectator';
import { AppComponent } from './app.component';
import { ComponentFixture } from '@angular/core/testing';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { BookListComponent } from './book-list/book-list.component';
import { provideMockStore } from '@ngrx/store/testing';
import { BookCollectionComponent } from './book-collection/book-collection.component';

describe('AppComponent', () => {
  let spectator: Spectator<AppComponent>;
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  const createComponent = createComponentFactory({
    component: AppComponent,
    declarations: [MockComponent(BookListComponent), MockComponent(BookCollectionComponent)],
    imports: [HttpClientTestingModule],
    providers: [provideMockStore({ initialState: { books: { books: [] }, collection: { selectedBooks: [] } } })],
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
    fixture = spectator.fixture;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
