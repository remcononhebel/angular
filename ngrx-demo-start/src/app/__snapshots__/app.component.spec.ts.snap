// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<app-root
  bookCollection$={[Function Store]}
  books$={[Function Store]}
  booksApiService={[Function BooksApiService]}
  numberOfAvailableBooks$={[Function Store]}
  numberOfBooksInCollections$={[Function Store]}
  store={[Function MockStore]}
  title={[Function String]}
>
  <div
    class="app"
  >
    <div
      class="app__available-books"
    >
      <h1>
        Available books (0)
      </h1>
      <app-book-list
        class="book-list"
      />
    </div>
    <div
      class="app__my-collection"
    >
      <h1>
        My Collection (No books selected)
      </h1>
      <app-book-collection
        class="book-collection"
      />
    </div>
  </div>
</app-root>
`;
